### PR TITLE
Codechange: Replace mishmash of types for widget index with WidgetID.

### PR DESCRIPTION
**Pull request by PeterN**
## Motivation / Problem

Widget indexes were stored as int, but often passed around as uint/uint8_t and casts.

## Description

Alias WidgetID to int, and use WidgetID consistently instead. This does not address the mishmash of variable names...

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, game_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

**Comment by LordAro** [comment]
How about a typedef ?

**Comment by PeterN** [comment]
> How about a typedef ?

Done... much larger commit though!

**Comment by PeterN** [comment]
But also I found some more `int` which need replacing...

File: src/window_gui.h [hunk 1]
```diff
@@ -264,21 +264,21 @@ struct Window : ZeroedMemoryAllocator {
 	NWidgetStacked *shade_select;    ///< Selection widget (#NWID_SELECTION) to use for shading the window. If \c nullptr, window cannot shade.
 	Dimension unshaded_size;         ///< Last known unshaded size (only valid while shaded).
 
-	int mouse_capture_widget;        ///< Widgetindex of current mouse capture widget (e.g. dragged scrollbar). -1 if no widget has mouse capture.
+	int mouse_capture_widget; ///< int of current mouse capture widget (e.g. dragged scrollbar). -1 if no widget has mouse capture.
```
Comment (rubidium42) [inline src/window_gui.h]: I guess this can become `WidgetID` too.

File: src/window_type.h [hunk 2]
```diff
@@ -10,6 +10,9 @@
 #ifndef WINDOW_TYPE_H
 #define WINDOW_TYPE_H
 
+/** Widget ID. */
+using WidgetID = int;
```
Comment (rubidium42) [inline src/window_type.h]: Maybe make it clear that negative WidgetIDs are special.
```suggestion
/**
 * Widget ID.
 * Even though the ID is signed, actual IDs must be non-negative.
 * Negative IDs are used for special cases, like denoting 'no widget'. 
 */
using WidgetID = int;
```

**Review by rubidium42 — APPROVED** [review]

**Pull request merged**